app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,13 @@
+class PostsController < ApplicationController
+  def index
+    # fetch all posts belonging to current user
+  end
+
+  def show
+    # fetch single post using params[:id]
+  end
+
+  def specific_user
+    # fetch post belonging to current user using params[:id]
+  end
```

Comment:
- [ ] Kindly remove this controller action as it is unnecessary. We only want the needed routes for now.

Reply:
I have resolved this issue. Thanks for the feedback